js/util-galery.js

```diff
@@ -43,27 +50,53 @@ const createComment = ({avatar, name, message}) => {
   return comment;
 };
 
+
 const renderComments = (comments) => {
+
+  numberOfCommentsShown += COMMENTS_PER_PORTION;
   socialComments.innerHTML = '';
+
+  if (numberOfCommentsShown >= comments.length) {
+    commentsLoader.classList.add('hidden');
+    numberOfCommentsShown = comments.length;
+  } else {
+    commentsLoader.classList.remove('hidden');
+  }
+
   const fragment = document.createDocumentFragment();
 
-  comments.forEach((comment) => {
-    fragment.append(createComment(comment));
-  });
+  for (let i = 0; i < numberOfCommentsShown; i++) {
+    fragment.append(createComment(comments[i]));
+  }
 
   socialComments.append(fragment);
+  commentCounter.innerHTML = `${numberOfCommentsShown} из <span class="comments-count">${comments.length}</span> комментариев`;
+  //console.log(numberOfCommentsShown);
 };
 
+function onCommentsLoaderButton (data) { // не могу придумать подходящее имя пораметру
+  renderComments(data);
+}
+
 const showBigPhoto = (modal) => {
   bigPhoto.classList.remove('hidden');
   document.body.classList.add('modal-open');
-  commentCounter.classList.add('hidden');
-  commentsLoader.classList.add('hidden');
+  // commentCounter.classList.add('hidden');
+  // commentsLoader.classList.add('hidden');
   document.addEventListener('keydown', onDocumentKeydown);
-  closeButton.addEventListener('click', onCloseButton); // пока не понимаю где удалять эти обработчики..
+  closeButton.addEventListener('click', onCloseButton);
 
   renderPhotoDetails(modal);
   renderComments(modal.comments);
+  commentsLoader.addEventListener('click', () => onCommentsLoaderButton(modal.comments));
```

Comment:
Смотри, ты вешаешь в качестве обработчика функцию `() => onCommentsLoaderButton(modal.comments)` Т.е. анонимная стрелочная функция внутри которой ты вызываешь функцию `onCommentsLoaderButton(modal.comments)`. Именно эту функцию тебе и нужно удалить при закрытии этого попапа, не  `commentsLoader.removeEventListener('click', onCommentsLoaderButton);` а именно вот эту анонимную, иначе ты её не удалишь и она продолжит рендерить данные с предыдущей фотографии так же по клику и при каждом открытии новой фотографии эти обработчики будут накладываться все больше и больше...
```ts
  commentsLoader.addEventListener('click', () => onCommentsLoaderButton(modal.comments));
```
Решений несколько, но идея в том, что нужно либо массив комментариев сделать глобальным, либо именно эту функцию, чтобы её удалить, либо отвязать функцию от аргументов.
Могу предложить такое вот решение:
```suggestion
  // Записываем эту нашу функцию в глобальную переменную (объявляем её там же, где и numberOfCommentsShown)
  onCommentsLoaderButton = () => renderComments(modal.comments);
  // И устанавливаем её в качестве обработчика
  commentsLoader.addEventListener('click', onCommentsLoaderButton);
```